### PR TITLE
Add image compare machinery for tests of plots

### PR DESCRIPTION
**Pull request by cdeil**
We should copy the image testing machinery from wcsaxes (which is based on matplotlib) to test the astroplan plotting functions @jberlanga is developing.

I don't know how difficult this is and how much work to set up, see https://github.com/astrofrog/wcsaxes/blob/6a304a33af78096e37d22c08e8ad1b27a76c8478/wcsaxes/tests/test_images.py#L43.
It should be done in an independent PR from the ones implementing the actual plotting routines, just using one dummy plot for now.

@astrofrog – Would it make sense to move some utility functions or classes and document how to do this in Astropy? (I want the same for Gammapy and there might be other affiliated package implementing plotting code and wanting to test it)

### CODE REVIEW

**Comment by cdeil** [comment]
@eteq or @adrn – Can you review this?

I'd like to merge this now, and leave the actual implementation of tests for the astroplan plotting functions to a future PR (maybe by me later today if I can find the time) after #65 has been fixed.

**Comment by cdeil** [comment]
Finally ... after fiddling around with pytest-mpl and travis-ci ... the example image comparison test is now running and passing on travis-ci:
https://travis-ci.org/astropy/astroplan/jobs/77686401#L626

**Comment by cdeil** [comment]
Travis-ci tests passed ... OK to merge?

**Comment by adrn** [comment]
@cdeil looks good to me!

File: astroplan/plots/tests/test_sky.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+# TODO: replace this with actual plot checks once these
+# issues are resolved:
+# https://github.com/astropy/astroplan/issues/65
+# https://github.com/astropy/astroplan/issues/74
+@pytest.mark.skipif('not HAS_MATPLOTLIB')
+@pytest.mark.mpl_image_compare
```
Comment (bsipocz) [inline astroplan/plots/tests/test_sky.py]: Wouldn't you need to give this an image name? Or this just works as there is only one image tested at the moment? (I just wonder as I'm trying to do this for photutils at the moment.)

Comment (cdeil) [inline astroplan/plots/tests/test_sky.py]: @bsipocz – I think it generates the image filename from the test name if no filename is given.
So I expect this to still work for multiple images / tests, but I didn't check.
Probably I'll start implementing image tests for Astroplan now.

Comment (bsipocz) [inline astroplan/plots/tests/test_sky.py]: Oh, that may solve my problem in photutils then. I've tried the to follow the ``wcsaxes`` example, but then stuck due to the parametrized tests.

**Pull request merged**